lib/utils/udpsender.js

```diff
@@ -0,0 +1,20 @@
+require('./KalturaConfig');
+
+var PORT = KalturaConfig.config.udpsender.PORT;
+var HOST = KalturaConfig.config.udpsender.HOST;
+var client = require('dgram').createSocket('udp4');
+
+var sendFunction = function (message,response){
+    var message_buffer = new Buffer(message);
+    response.log('#UDP ' + message_buffer + ' sent to ' + HOST + ':' + PORT);
+    client.send(message_buffer, 0, message_buffer.length,PORT, HOST,
+        function (err, bytes) {
+            if (err) {
+                client.close();
+            }

```

Comment:
So I'm agree and not sure why udp was used to begin with here

Sent from my iPhone

On 14 ????? 2016, at 9:47, nirbe <notifications@github.com<mailto:notifications@github.com>> wrote:


In lib/utils/udpsender.js<https://github.com/kaltura/play-server/pull/244#discussion_r59669655>:

> @@ -0,0 +1,20 @@
> +require('./KalturaConfig');
> +
> +var PORT = KalturaConfig.config.udpsender.PORT;
> +var HOST = KalturaConfig.config.udpsender.HOST;
> +var client = require('dgram').createSocket('udp4');
> +
> +var sendFunction = function (message,response){
> +    var message_buffer = new Buffer(message);
> +    response.log('#UDP ' + message_buffer + ' sent to ' + HOST + ':' + PORT);
> +    client.send(message_buffer, 0, message_buffer.length,PORT, HOST,
> +        function (err, bytes) {
> +            if (err) {
> +                client.close();
> +            }


@etameran<https://github.com/etameran> - I agree UDP is light wight - however the motivation for this change is not to receive real-times alerts but to be able to answer beacon/fill-rate differences questions faster and in more convenient way. No need for real-time at all. Also this depends on how well the UDP client works - if significant (yet small in absolute numbers) number of logs will fail to deliver - we'll end up looking into raw logs again to make sure the numbers we provide are accurate.

Another concern is that errors/exceptions of this sender will interfere with the normal process behavior - so the erroneous cases should be tested,

Last there's no need to parse anything, even if using external process: we can write the log messages in JSON format, grep for these special logs only and send them over (UDP or any other method)

-
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/kaltura/play-server/pull/244/files/5df3080aa6fb29f7523f4380a2c3060f0865ff15#r59669655>